dashboard/app/models/script_level.rb

```diff
@@ -94,7 +94,7 @@ def next_level_or_redirect_path_for_user(user)
     # to unplugged level (example: if you start a sequence of
     # assessments associated with an unplugged level you should
     # continue on that sequence instead of skipping to next stage)
-    level_to_follow = valid_progression_level? ? next_progression_level : next_level
+    level_to_follow = unplugged? ? next_level : next_progression_level(user)
```

Comment:
You are right. The situation I was trying to avoid was that you are on the last level of a stage, which happens to be unplugged. The next stage happens to be hidden. We should still skip over the hidden stage.

Really we want to know in what way the level is not "valid", and only continue to the next level if it is invalid in the same way. I'll think about how to better accomplish this.